### DISCUSSION TITLE
Problemas con la asociación  de las imágenes a la hora de crear el formulario

### DISCUSSION THREAD

**Discussion by ivasansan1** [Aparking/AparKing_Backend#105] [Show and tell]
Estoy teniendo problemas con asociar la imagen, o me da fallo del core, o no se guarda el fumulario de la imagen, o da error de formulario etc. he probado por dos vias:

### 1. hacerlo todo junto y que el id sea el mismo que cuando se crea (no me sale)
En la imagen se ve como se crea el formulario con la parte de la imagen, pero esta no se guarda ni asocia al id del garaje
![image](https://github.com/Aparking/AparKing_Backend/assets/95249149/416716a4-f7fe-4538-a993-eb6413fd01c7)

Aquí está como si se ha creado y su id es el  25
![image](https://github.com/Aparking/AparKing_Backend/assets/95249149/7d0653d4-ce08-49ec-a72c-6e1d1b635837)

Pero no se ha guardado ninguna imagen con el id 25 (lo otro son pruebas que yo he hecho)
![image](https://github.com/Aparking/AparKing_Backend/assets/95249149/33930cd7-878d-498a-b2f2-423def27aaa0)

### 2. crear el garaje, y posteriormente crear la imagen cuyo id se garaje se optine del ultimo de la lista de garajes (nada sale error de core)

En esta imagen esta el formulario de imágenes separado al cual se le da el id pero da fallo del core
![image](https://github.com/Aparking/AparKing_Backend/assets/95249149/dd7bfec0-12a1-4d0c-a8a6-602daf78456e)

Aquí se ve como se guarda la info del fomulario
![image](https://github.com/Aparking/AparKing_Backend/assets/95249149/ec4d86f9-ea9e-493b-80cf-2ac51d7500ee)

-------------------------------

Y si, para el core he ya leí la otra discusión, he probado con google Chrome, OperaGX, Firefox, Egde y nada

**Comment by perealberto** [comment]
Buenas tardes Iván. Dos cosas:

1. En cuanto a las imágenes en el formulario: creo que según como está el proyecto construido puedes hacer dos cosas.
- Manejar el orden de creación desde el frontend. Es decir, recoger la información de creación del formulario y en el método de creación del garaje desde el frontend primero mandas una petición de creación del garaje al backend y cuando se haya creado, mandas las peticiones de creación de imágenes junto con el id del garaje creado.
- Editar el backend y cambiar la función de create() de un garaje, haciendo que maneje los datos que recibe directamente del formulario del frontend y cree allí todo en el orden correcto. Personalmente creo que esta es la más sencilla.

2. En cuanto al problema del CORS, ¿probaste realizar el cambio de IP en el servicio del frontend? [Está explicado en este comentario](https://github.com/Aparking/AparKing_Backend/discussions/75#discussioncomment-8697489).

**Reply by ivasansan1** [reply]
Buenos días @3ralon. 

Ayer seguí su explicación para solucionar el CORS pero no obtuve éxito. Voy a probar a modificar el bakend tal como comentas en el segundo punto del apartado 1. Creo que si lo hago todo directamente no me dará fallos. Esta tarde estaré en el servidor con ello para que pueda continuar con su tarea.
